chat-plugins/othermetas.js

```diff
@@ -0,0 +1,106 @@
+		};
+		for (let statId in megaTemplate.baseStats) {
+			deltas.baseStats[statId] = megaTemplate.baseStats[statId] - baseTemplate.baseStats[statId];
+		}
+		if (megaTemplate.types.length > baseTemplate.types.length) {
+			deltas.type = megaTemplate.types[1];
+		} else if (megaTemplate.types.length < baseTemplate.types.length) {
+			deltas.type = baseTemplate.types[0];
+		} else if (megaTemplate.types[1] !== baseTemplate.types[1]) {
+			deltas.type = megaTemplate.types[1];
+		}
+		//////////////////////////////////////////
+		let ability = deltas.ability;
+		let types = template.types;
+		let baseStats = Object.assign({}, template.baseStats);
+		if (types[0] === deltas.type) { // Add any type gains
+			types = [deltas.type];
+		} else if (deltas.type) {
+			types = [types[0], deltas.type];
+		}
+		for (let statName in baseStats) { // Add the changed stats and weight
+			baseStats[statName] = Tools.clampIntRange(baseStats[statName] + deltas.baseStats[statName], 1, 255);
+		}
+		let weightkg = Math.round(Math.max(0.1, template.weightkg + deltas.weightkg) * 100) / 100;
+		let type = '<span class="col typecol">';
+		for (let i = 0; i < types.length; i++) { // HTML for some nice type images.
+			type = `${type}<img src="https://play.pokemonshowdown.com/sprites/types/${types[i]}.png" alt="${types[i]}" height="14" width="32">`;
+		}
+		type = type + "</span>";
+		let gnbp = 20;
+		if (weightkg >= 200) { // Calculate Grass Knot/Low Kick Base Power
+			gnbp = 120;
+		} else if (weightkg >= 100) {
+			gnbp = 100;
+		} else if (weightkg >= 50) {
+			gnbp = 80;
+		} else if (weightkg >= 25) {
+			gnbp = 60;
+		} else if (weightkg >= 10) {
+			gnbp = 40;
+		} // Aah, only if `template` had a `bst` property.
+		let bst = baseStats['hp'] + baseStats['atk'] + baseStats['def'] + baseStats['spa'] + baseStats['spd'] + baseStats['spe'];
+		let text = `<b>Stats</b>: ${Object.values(baseStats).join('/')}<br />`;
+		text = `${text}<b>BST</b>:${bst}<br />`;
+		text = `${text}<b>Type:</b> ${type}<br />`;
+		text = `${text}<b>Ability</b>: ${ability}<br />`;
+		text = `${text}<b>Weight</b>: ${weightkg} kg (${gnbp} BP)`;
+		return this.sendReplyBox(text);
+	},
+	mixandmegahelp: ["/mnm <pokemon> @ <mega stone> - Shows the mix and mega evolved Pokemon's type and stats."],
+
+	'350' : 'cup350',
+	cup350 : function (target, room, user) {
+		if (!this.runBroadcast()) return;
+		if (!Tools.data.Pokedex[toId(target)]) {
+			return this.errorReply("Error: Pokemon not found.");
+		}
+		let bst = 0, pokeobj = Tools.getTemplate(toId(target));
+		for (let i in pokeobj.baseStats) {
+			bst += pokeobj.baseStats[i];
+		}
+		let newStats = {};
+		for (let i in pokeobj.baseStats) {
+			newStats[i] = pokeobj.baseStats[i] * (bst <= 350 ? 2 : 1);
+		}
+		let text = `${pokeobj.species} in 350 Cup: <br /> ${Object.values(newStats).join('/')}`;
+		this.sendReplyBox(text);
+	},
+	cup350help: ["/350 <pokemon> - Shows the base stats that a Pokemon would have in 350 cup."],
```

Comment:
This should probably be ``/cup350 <pokemon>...`` - instead of using the alias in the help command.  Alternatively, you could also say ``/cup250 OR /350 <pokemon>``... or something to that effect